pkg/operator/configobserver/config_observer_controller.go

```diff
@@ -135,6 +171,46 @@ func (c ConfigObserver) sync(ctx context.Context, syncCtx factory.SyncContext) e
 	return configError
 }
 
+func (c ConfigObserver) updateObservedConfig(syncCtx factory.SyncContext, existingConfig map[string]interface{}, mergedObservedConfig map[string]interface{}) error {
+	if len(c.nestedConfigPath) == 0 {
+		if !equality.Semantic.DeepEqual(existingConfig, mergedObservedConfig) {
+			syncCtx.Recorder().Eventf("ObservedConfigChanged", "Writing updated observed config: %v", diff.ObjectDiff(existingConfig, mergedObservedConfig))
+			return c.updateConfig(syncCtx, mergedObservedConfig)
+		}
+		return nil
+	}
+
+	existingConfigNested, _, err := unstructured.NestedMap(existingConfig, c.nestedConfigPath...)
+	if err != nil {
+		return fmt.Errorf("unable to extract the config under %v key, err %v", c.nestedConfigPath, err)
+	}
+	mergedObservedConfigNested, _, err := unstructured.NestedMap(mergedObservedConfig, c.nestedConfigPath...)
+	if err != nil {
+		return fmt.Errorf("unable to extract the merged config under %v, err %v", c.nestedConfigPath, err)
+	}
+	if !equality.Semantic.DeepEqual(existingConfigNested, mergedObservedConfigNested) {
+		syncCtx.Recorder().Eventf("ObservedConfigChanged", "Writing updated section (%q) of observed config: %q", strings.Join(c.nestedConfigPath, "/"), diff.ObjectDiff(existingConfigNested, mergedObservedConfigNested))
+
+		updatedConfig := existingConfig
+		if err := unstructured.SetNestedField(updatedConfig, mergedObservedConfigNested, c.nestedConfigPath...); err != nil {
```

Comment:
You should only modify your values to prevent races.

At this point, `existingConfig` may already contain obsolete values for all the other observers running in parallel with you, wheareas these other observers might already be in a new cycle where their `existingConfig` may contain obsolete values to your new config.

I think you should create a new function based on `type UpdateOperatorSpecFunc func(spec *operatorv1.OperatorSpec) error` that just takes the observed config and only replaces its managed prefixed config.

Reply:
yeah, I can't write only my changes because that would remove everything else. I thought that the potential conflicts would be automatically removed - PATH - will check.